src/item.cpp

```diff
@@ -1619,6 +1619,60 @@ bool _stacks_components( item const &lhs, item const &rhs, bool check_components
            ( lhs.get_uncraft_components() == rhs.get_uncraft_components() );
 }
 
+bool _stacks_custom_item_prefix( item const &lhs, item const &rhs )
+{
+    std::set<flag_id> prefix1;
+    for( const flag_id &f : lhs.get_flags() ) {
```

Comment:
This is going to be pretty slow and you're also iterating through the flags twice (three times when you consider `tname::segments::TAGS`). This should be merged with `_stacks_custom_item_suffix()` so it only iterates once and returns a pair of bools.

<details>
<summary>Quick perf test</summary>

Debug timer for opening `pickup all` from this save with an -O3 build:  [Black Gap-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/14237824/Black.Gap-trimmed.tar.gz)

Before:
![0 - before PR](https://github.com/CleverRaven/Cataclysm-DDA/assets/68240139/9f512b8a-cf2e-471a-a05c-fa491893bbb3)

After:
![1 - with PR](https://github.com/CleverRaven/Cataclysm-DDA/assets/68240139/d8fec9b2-8e79-48f6-a52f-f442d985d262)

None of these items have these special flags so there's a non-trivial perf impact for no benefit.

</details>


Can't these special flags be cached separately?

I'm not sure if these labels are important or common enough to warrant aggregation. If performance can't be improved, consider setting the bits when `tname::segments::TAGS` is set instead of running a separate check.

Reply:
Maybe I can create `cata::heap<FlagsSetType> prefix_tags_cache` and `cata::heap<FlagsSetType> suffix_tags_cache` ?